packages/utilities/src/classNamesFunction.test.ts

```diff
@@ -0,0 +1,43 @@
+import { classNamesFunction } from './classNamesFunction';
+import { Stylesheet } from '@uifabric/merge-styles';
+
+describe('classNamesFunction', () => {
+  let lastRule: string;
+
+  beforeEach(() => {
+    Stylesheet.getInstance().setConfig({
+      onInsertRule: () => {
+        /*no-op*/
+      }
+    });
+  });
+
+  afterEach(() => {
+    Stylesheet.getInstance().setConfig({
+      onInsertRule: undefined
+    });
+  });
+
+  it('can cache rules', () => {
+    let styleFunctionCalled = false;
+
+    const getClassNames = classNamesFunction();
+    const getStyles = (props: { a: number }) => {
```

Comment:
should really just do `jest.fn.mockImplementation((props: ... => { ...})` here for getStyles, then you can get rid of styleFunctionCalled and do the far more idomatic `expect(getStyles).toHaveBeenCalledTimes(1)` below in line 33 and 36 (and in turn, `toHaveBeenCalledTimes(2)` in line 38 and 41)

Reply:
thanks, will try it out!